### DISCUSSION TITLE
Will the new ZIP-feature also work on self-extracting (exe) ZIP-archives?

### DISCUSSION THREAD

**Discussion by borque** [microsoft/winget-pkgs#88247] [Q&A] [answered]
Hello!
Some programs are delivered as self-extracting ZIP-archives that automatically run some sort of setup program after extraction into a temp folder. Arguments (for unattended silent installation) specified when running the self-extractor are not passed through to the final setup program.
So my question is, whether wingets upcoming new ability to work with ZIP archives can also be used on these cases of self-extracting ZIP-archives to make them usable by winget?
One example for this behavior where the nested setup.exe can only be silenced be running it directly with proper arguments ist Fineprint: https://fineprint.com/release/fp1131.exe
Or is there any other solution I miss?
Thanks!

**Comment by ryfu-msft** [comment, marked as answer]
Hey @borque,

No, winget's ZIP-feature will not be able to handle the scenario you described above. If you wanted to be able to properly pass the arguments to the setup program, you would need to provide an installerURL that points to the zip archive containing the files that you want to extract rather than a self-extracting exe file. Winget will then handle the extraction process and pass the arguments directly to the nested installer that you specify in the manifest.

**Comment by borque** [comment]
Hello!
Thanks for your reply, @ryfu-msft.
I have created issue #89226 for this specific installer "Finprint". I think further discussion might be better off there, as this could also be a general issue (SFX-ZIPs)?
